samples/boards/nrf52/mesh/onoff_level_lighting_vnd_app/src/mesh/device_composition.h

```diff
@@ -10,26 +10,34 @@
 
 #define CID_ZEPHYR 0x0002
 
-enum lightness { ONPOWERUP = 0x01, ONOFF, LEVEL, ACTUAL, LINEAR, CTL, IGNORE};
-enum temperature { ONOFF_TEMP = 0x01, LEVEL_TEMP, CTL_TEMP, IGNORE_TEMP};
+/* Refer 7.2 of Mesh Model Specification */
+#define RANGE_SUCCESSFULLY_UPDATED	0x00
+#define CANNOT_SET_RANGE_MIN		0x01
+#define CANNOT_SET_RANGE_MAX		0x02
+
+enum lightness { ONPOWERUP = 0x01, ONOFF, LEVEL, ACTUAL, LINEAR, CTL, IGNORE };
+enum temperature { ONOFF_TEMP = 0x01, LEVEL_TEMP, CTL_TEMP, IGNORE_TEMP };
 
 struct generic_onoff_state {
 	u8_t onoff;
-	u8_t previous;
+	u8_t current;
```

Comment:
I see you already have `onoff` in this struct? Isn't that the current state? At least the "get" message handler uses it for the response. What exactly is `current` then if it's not the current value?

Reply:
If I understand it right (looking more closely at the code than the last time - I wish you would have corrected me then!), the only reason you're not using a local variable is that you need a way to find out the results of calling `gen_onoff_set_unack()` from `gen_onoff_set()`. One solution could be to consider refactoring these functions so that you have a third function that is essentially the same as the `unack()` except that it's able to return an error in case of invalid parameters. You can then use the error in the non-unack case to decide whether to send the status or not.

Reply:
I will make it val

Reply:
That does not solve the issue I pointed out. As far as I see you don't need this value for anything else than the `set_unack()` -> `set()` call path, so having it in the state struct is overkill (and confusing since it gives the impression that it's needed in some wider context). Please correct me if I'm wrong with these observations.